### PR TITLE
Implement Hugging Face Image Embedding MLTransform

### PR DESCRIPTION
**Pull request by jrmccluskey**
Adds support for generating image embeddings via the sentence-transformers library. 

Part of #31500

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/python/apache_beam/ml/transforms/embeddings/huggingface.py [hunk 1]
```diff
@@ -153,6 +154,45 @@ def get_ptransform_for_processing(self, **kwargs) -> beam.PTransform:
...
+    MLTransform to embed image data. Models are loaded using the RunInference
+    PTransform with the help of ModelHandler.
+
+    Args:
+      model_name: Name of the model to use. The model should be hosted on
+        HuggingFace Hub or compatible with sentence_transformers. See
+        https://www.sbert.net/docs/sentence_transformer/pretrained_models.html#image-text-models # pylint: disable=line-too-long
+        for a list of sentence_transformers models.
+      columns: List of columns to be embedded.
+      min_batch_size: The minimum batch size to be used for inference.
```
Comment (liferoad) [inline sdks/python/apache_beam/ml/transforms/embeddings/huggingface.py:170]: shall we add all these options to `__init__`? similarly to `SentenceTransformerEmbeddings`?

File: sdks/python/apache_beam/ml/transforms/embeddings/huggingface.py [hunk 2]
```diff
@@ -153,6 +154,45 @@ def get_ptransform_for_processing(self, **kwargs) -> beam.PTransform:
...
+    return _SentenceTransformerModelHandler(
+        model_class=SentenceTransformer,
+        model_name=self.model_name,
+        load_model_args=self.load_model_args,
+        min_batch_size=self.min_batch_size,
+        max_batch_size=self.max_batch_size,
+        large_model=self.large_model)
+
+  def get_ptransform_for_processing(self, **kwargs) -> beam.PTransform:
+    # wrap the model handler in a _TextEmbeddingHandler since
```
Comment (liferoad) [inline sdks/python/apache_beam/ml/transforms/embeddings/huggingface.py:187]: `_TextEmbeddingHandler`?
and do we need to create `SentenceTransformerImageEmbeddings`? shall we just add `image_model:bool` in `SentenceTransformerEmbeddings`? or can we infer the model type automatically and then call either `_ImageEmbeddingHandler` or `_TextEmbeddingHandler`?

Comment (jrmccluskey) [reply sdks/python/apache_beam/ml/transforms/embeddings/huggingface.py:187]: Swapping to a bool assignment may be cleaner (and is how we probably need to handle the Inference API version as well,) let me take a run at writing that real quick

File: sdks/python/apache_beam/ml/transforms/embeddings/huggingface.py (hunk 1 above)
Comment (jrmccluskey) [reply sdks/python/apache_beam/ml/transforms/embeddings/huggingface.py]: Yes this wound up looking cleaner, PTAL

**Review by liferoad — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**